cmd/kubebuilder/initproject/project.go

```diff
@@ -0,0 +1,135 @@
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/util"
+	"sigs.k8s.io/controller-tools/pkg/scaffold"
+	"sigs.k8s.io/controller-tools/pkg/scaffold/input"
+	"sigs.k8s.io/controller-tools/pkg/scaffold/manager"
+	"sigs.k8s.io/controller-tools/pkg/scaffold/project"
+)
+
+var prj *project.Project
+var bp *project.Boilerplate
+var gopkg *project.GopkgToml
+var mrg *manager.Cmd
+var dkr *manager.Dockerfile
+
+// ProjectCmd represents the project command
+var ProjectCmd = &cobra.Command{
+	Use:   "init",
+	Short: "Scaffold a new project.",
+	Long: `Scaffold a project.
+
+Writes the following files:
+- a boilerplate license file
+- a PROJECT file with the domain and repo
+- a Makefile to build the project
+- a Gopkg.toml with project dependencies
+- a cmd/manager/main.go to run
+
+project will prompt the user to run 'dep ensure' after writing the project files.
+`,
+	Example: `# Scaffold a project using the apache2 license with "The Kubernetes authors" as owners
+kubebuilder init --domain k8s.io --license apache2 --owner "The Kubernetes authors"
+`,
+	Run: RunInit,
+}
+func RunInit(cmd *cobra.Command, args []string) {
+	// project and boilerplate must come before main so the boilerplate exists
+	s := &scaffold.Scaffold{
+		BoilerplateOptional: true,
+		ProjectOptional:     true,
+	}
+
+	p, _ := prj.GetInput()
+	b, _ := bp.GetInput()
+	err := s.Execute(input.Options{ProjectPath: p.Path, BoilerplatePath: b.Path}, prj, bp)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	s = &scaffold.Scaffold{}
+	err = s.Execute(input.Options{ProjectPath: p.Path, BoilerplatePath: b.Path},
+		gopkg, mrg, &project.Makefile{}, dkr, &manager.APIs{}, &manager.Controller{}, &manager.Config{},
+		&project.GitIgnore{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Run `dep ensure` to fetch dependencies (Recommended) [y/n]?")
+	if util.Yesno() {
+		c := exec.Command("dep", "ensure") // #nosec
+		c.Stderr = os.Stderr
+		c.Stdout = os.Stdout
+		fmt.Println(strings.Join(c.Args, " "))
+		if err := c.Run(); err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Println("Running make...")
+		c = exec.Command("make") // #nosec
+		c.Stderr = os.Stderr
+		c.Stdout = os.Stdout
+		fmt.Println(strings.Join(c.Args, " "))
+		if err := c.Run(); err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		fmt.Println("Skipping `dep ensure`.  Dependencies will not be fetched.")
+	}
+}
+
+func AddProjectCommand(cmd *cobra.Command) {
+	cmd.AddCommand(ProjectCmd)
+
+	prj = ProjectForFlags(ProjectCmd.Flags())
+	bp = BoilerplateForFlags(ProjectCmd.Flags())
+	gopkg = &project.GopkgToml{}
+	mrg = &manager.Cmd{}
+	dkr = &manager.Dockerfile{}
+}
+
+// ProjectForFlags registers flags for Project fields and returns the Project
+func ProjectForFlags(f *flag.FlagSet) *project.Project {
+	p := &project.Project{}
+	f.StringVar(&p.Domain, "project-domain", "k8s.io", "Works only with --newkb, domain for groups")
```

Comment:
Lets reuse the domain flag that is already defined.

Reply:
Done